cli/pkg/kctrl/cmd/package/installed/create_or_update.go

```diff
@@ -334,14 +350,23 @@ func (o CreateOrUpdateOptions) update(client kubernetes.Interface, kcClient kccl
 		}
 	}
 
-	o.statusUI.PrintMessagef("Updating package install for '%s' in namespace '%s'", o.Name, o.NamespaceFlags.Name)
-	o.addCreatedResourceAnnotations(&updatedPkgInstall.ObjectMeta, false, isSecretCreated, isSecretDeleted)
-	_, err = kcClient.PackagingV1alpha1().PackageInstalls(o.NamespaceFlags.Name).Update(
-		context.Background(), updatedPkgInstall, metav1.UpdateOptions{},
-	)
-	if err != nil {
-		err = fmt.Errorf("Updating package '%s': %s", o.Name, err.Error())
-		return err
+	if isSecretCreated || changed {
+		o.statusUI.PrintMessagef("Updating package install for '%s' in namespace '%s'", o.Name, o.NamespaceFlags.Name)
+		o.addCreatedResourceAnnotations(&updatedPkgInstall.ObjectMeta, false, isSecretCreated, isSecretDeleted)
+		_, err = kcClient.PackagingV1alpha1().PackageInstalls(o.NamespaceFlags.Name).Update(
+			context.Background(), updatedPkgInstall, metav1.UpdateOptions{},
+		)
+		if err != nil {
+			err = fmt.Errorf("Updating package '%s': %s", o.Name, err.Error())
+			return err
+		}
+	}
+
+	if reconciliationPaused {
+		err = o.unpauseReconciliation(kcClient)
```

Comment:
Pausing and unpausing if needed at the beginning and the end is the best way I could find. Maybe we should start logging pauses and unpauses, so that if things error out in the middle the user is aware that reconciliation is paused?
Not sure if it makes sense to defer a function which unpauses the app if paused :/

Reply:
keeping it paused it probably good (since they may want to keep on reconfiguring packageinstall) but good thought on indicating that to the user.